src/main/java/net/fabricmc/loom/extension/MixinExtensionApiImpl.java

```diff
@@ -50,10 +50,10 @@ public abstract class MixinExtensionApiImpl implements MixinExtensionAPI {
 	public MixinExtensionApiImpl(Project project) {
 		this.project = Objects.requireNonNull(project);
 		this.useMixinAp = project.getObjects().property(Boolean.class)
-				.convention(project.provider(() -> !LoomGradleExtension.get(project).isNeoForge()));
+				.convention(project.provider(() -> !LoomGradleExtension.get(project).isNeoForge() && (!LoomGradleExtension.get(project).isForge() || !LoomGradleExtension.get(project).getForgeProvider().usesMojangAtRuntime())));
```

Comment:
The logic here is a bit ugly, it could be better for readability to use a block lambda.

@shedaniel Do you remember if there's any reason why we have this set to false for Fabric? Upstream 1.6 has it on true: https://github.com/FabricMC/fabric-loom/blob/c4d36fac4ea7ccd1ef9526aa138139a154c8f581/src/main/java/net/fabricmc/loom/extension/MixinExtensionApiImpl.java#L51-L52
(possibly for common projects that need their refmap to be transformed? But that's not needed if refmaps are removed altogether)

Reply:
I believe old forge mods should have this as false

Reply:
Yeah, that's fine. I think we could change it for Fabric if it didn't break common projects not getting refmaps on old Forge (arch plugin could set it to true?)